datasize_macro/src/lib.rs

```diff
@@ -0,0 +1,72 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use impl_data_sizes::{
+	impl_datasize, impl_static_data_size, retrieve_generics, retrieve_lifetimes,
+};
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, DeriveInput};
+
+mod impl_data_sizes;
+
+#[proc_macro_derive(DataSize)]
+pub fn derive_data_size(item: TokenStream) -> TokenStream {
+	let input = parse_macro_input!(item as DeriveInput);
+	let ident = &input.ident;
+	let inner = impl_datasize(&input);
+
+	// Parsing generics and lifetimes
+	let generics = retrieve_generics(&input);
+	let lifetimes = retrieve_lifetimes(&input);
+	// Here we check if there is any lifetime AND generics
+	// if there is both, we need a comma to separate them
+	let generics_and_lifetimes = if generics.len() > 0 && lifetimes.len() > 0 {
+		quote!(<#(#lifetimes),*, #(#generics)*>)
+	} else {
+		quote!(<#(#lifetimes),* #(#generics)*>)
+	};
+
+	let output = quote! {
+		impl<#(#generics: DataSize),*> cornflakes::datasize::DataSize for #ident #generics_and_lifetimes {
```

Comment:
It seems like I can't make a bound to `DataSize` or `DataSize + StaticDataSize` for type generics since I can't modify the content of `impl_generics` (the field it contain is private).
even `#(#impl_generics: DataSize),*` do not work, sadly (but it was expected).

Reply:
You modify it before you split it - you can push bounds to `Generics`.  See the example in `syn` for how to do that, I forgot lol

Reply:
Ok I haven't seen your reply, working in this rn